tests/integration/p2p.go

```diff
@@ -140,6 +140,9 @@ func connectPeers(
 	nodes []*node.Node,
 	addresses []string,
 ) chan struct{} {
+	// If we have some database actions prior to connecting the peers, we want to ensure that they had time to
+	// complete before we connect. Otherwise we might wrongly catch them in our wait function.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
praise: Nice spot - I didn't notice this looking through your wip branch earlier :)